delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/ResolversDummy.scala

```diff
@@ -0,0 +1,136 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.Resolvers._
+import ch.epfl.bluebrain.nexus.delta.sdk._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{Project, ProjectRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.ResourceSourceParser._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverCommand.{CreateResolver, DeprecateResolver, TagResolver, UpdateResolver}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverRejection._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverState.Initial
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, IdSegment, Label}
+import ch.epfl.bluebrain.nexus.delta.sdk.testkit.ResolversDummy.ResolverJournal
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.UUIDF
+import ch.epfl.bluebrain.nexus.testkit.IOSemaphore
+import monix.bio.{IO, Task, UIO}
+
+/**
+  * A dummy Resolvers implementation
+  *
+  * @param journal     the journal to store events
+  * @param projects    the projects operations bundle
+  * @param semaphore   a semaphore for serializing write operations on the journal
+  */
+class ResolversDummy private (journal: ResolverJournal, projects: Projects, semaphore: IOSemaphore)(implicit
+    clock: Clock[UIO],
+    uuidF: UUIDF
+) extends Resolvers {
+
+  override def create(projectRef: ProjectRef, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource] =
+    eval(None, projectRef, resolverFields.id) { iri =>
+      CreateResolver(iri, projectRef, resolverFields.value)
+    }
+
+  override def create(id: IdSegment, projectRef: ProjectRef, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource] =
+    eval(Some(id), projectRef, resolverFields.id) { iri =>
+      CreateResolver(iri, projectRef, resolverFields.value)
+    }
+
+  override def update(id: IdSegment, projectRef: ProjectRef, rev: Long, resolverFields: ResolverFields)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource] =
+    eval(Some(id), projectRef, resolverFields.id) { iri =>
+      UpdateResolver(iri, projectRef, resolverFields.value, rev)
+    }
+
+  override def tag(id: IdSegment, projectRef: ProjectRef, tag: Label, tagRev: Long, rev: Long)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource] =
+    eval(Some(id), projectRef) { iri =>
+      TagResolver(iri, projectRef, tagRev, tag, rev)
+    }
+
+  override def deprecate(id: IdSegment, projectRef: ProjectRef, rev: Long)(implicit
+      caller: Caller
+  ): IO[ResolverRejection, ResolverResource] =
+    eval(Some(id), projectRef) { iri =>
+      DeprecateResolver(iri, projectRef, rev)
+    }
+
+  override def fetch(id: IdSegment, projectRef: ProjectRef): IO[ResolverRejection, Option[ResolverResource]] =
+    for {
+      p   <- projects.fetchFromCache(projectRef)
```

Comment:
why do we need to fetch from cache here instead of simply fetch?

Reply:
For performance ?

Reply:
Didn't notice this was a dummy, please disregard this one; the impls would need to be adjusted, though.